File: JavaScript/Exams/CSharpBrackets.js
function Solve(args) {
    var lines = parseInt(args[0]);
    var indentation = args[1];
    var formatted = "";
    var currentContent = "";
    var indentationsCount = 0;
    for (var i = 0; i < lines; i++) {
        var currentLine = args[i + 2].trim();
        for (var j = 0; j < currentLine.length; j++) {
            if (currentLine[j] == "{") {
                formatted += currentContent.trim();
                currentContent = "";
                formatted += "\n";
                formatted += appendIndentations(indentation, indentationsCount);
                formatted += "{\n";
                indentationsCount++;
            }
            else if (currentLine[j] == "}") {
                formatted += currentContent.trim();
                currentContent = "";
                formatted += "\n";
                indentationsCount--;
                formatted += appendIndentations(indentation, indentationsCount);
                formatted += "}\n";
            }
            else {
                if (formatted[formatted.length - 1] == "\n") {
                    formatted += appendIndentations(indentation, indentationsCount);
                }
                if (currentContent[currentContent.length - 1] == " " && currentLine[j] == " ") {
                    continue;
                }
                currentContent += currentLine[j];
            }
        }
        formatted += currentContent.trim();
        currentContent = "";
        formatted += "\n";
    }
    return getResult(formatted);
}

function getResult(formatted) {
    var result = formatted[0];
    for (var i = 1; i < formatted.length; i++) {
        if (formatted[i] == "\n" && formatted[i - 1] == "\n") {
            continue;
        }
        result += formatted[i];
    }
    return result.trim();
}

function appendIndentations(indentation, indentationsCount) {
    var result = "";
    for (var i = 0; i < indentationsCount; i++) {
        result += indentation;
    }
    return result;
}

File: C# II Projects/CSharpExam2 - Day2/CSharpBrackets/CSharpBrackets.cs
using System;
using System.Text;

class CSharpBrackets
{
    static void Main()
    {
        int linesCount = int.Parse(Console.ReadLine());
        string indentation = Console.ReadLine();
        StringBuilder tempFormatted = new StringBuilder();
        StringBuilder currentContent = new StringBuilder();
        int indentationsCount = 0;
        for (int i = 0; i < linesCount; i++)
        {
            string currentLine = Console.ReadLine().Trim();
            for (int j = 0; j < currentLine.Length; j++)
            {
                if (currentLine[j] == '{')
                {
                    tempFormatted.Append(currentContent.ToString().Trim());
                    currentContent = new StringBuilder();
                    tempFormatted.AppendLine();
                    tempFormatted.Append(AddIndentations(indentation, indentationsCount));
                    tempFormatted.AppendLine("{");
                    indentationsCount++;
                }
                else if (currentLine[j] == '}')
                {
                    tempFormatted.Append(currentContent.ToString().Trim());
                    currentContent = new StringBuilder();
                    tempFormatted.AppendLine();
                    indentationsCount--;
                    tempFormatted.Append(AddIndentations(indentation, indentationsCount));
                    tempFormatted.AppendLine("}");
                }
                else
                {
                    if (tempFormatted.Length > 0 && tempFormatted[tempFormatted.Length - 1] == '\n')
                    {
                        tempFormatted.Append(AddIndentations(indentation, indentationsCount));
                    }
                    if (currentContent.Length > 0 && currentContent[currentContent.Length - 1] == ' ' && currentLine[j] == ' ')
                    {
                        continue;
                    }
                    currentContent.Append(currentLine[j]);    
                }
            }
            tempFormatted.Append(currentContent.ToString().Trim());
            currentContent = new StringBuilder();
            tempFormatted.AppendLine();
        }
        Console.WriteLine(GetFormattedCode(tempFormatted.ToString()));
    }

    static string GetFormattedCode(string tempFormatted)
    {
        StringBuilder formatted = new StringBuilder();
        string[] splittedTempFormatted = tempFormatted.Split(new string[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
        for (int i = 0; i < splittedTempFormatted.Length; i++)
        {
            formatted.AppendLine(splittedTempFormatted[i]);
        }
        formatted.Length--;
        return formatted.ToString();
    }

    static string AddIndentations(string indentation, int count)
    {
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < count; i++)
        {
            result.Append(indentation);
        }
        return result.ToString();
    }
}
